lib/CL/pocl_llvm_utils.cc

```diff
@@ -152,14 +153,43 @@ char *get_cpu_name() {
 
 // TODO this should be fixed to not require LLVM eventually,
 // so that LLVM-less builds also report FMA correctly.
-#ifdef OCS_AVAILABLE
 int cpu_has_fma() {
   StringMap<bool> features;
   bool res = llvm::sys::getHostCPUFeatures(features);
   assert(res);
   return ((features["fma"] || features["fma4"]) ? 1 : 0);
 }
-#endif
+
+#define VECWIDTH(x)                                                            \
+  std::min(std::max((lane_width / (unsigned)(sizeof(x))), 1U), 16U)
+
+void cpu_setup_vector_widths(cl_device_id dev) {
```

Comment:
I think this belongs to the same place directory where the hwloc code is. You use LLVM's infra for detecting vector types for basic/pthread only as it works only with the host CPU, right?

Reply:
> You use LLVM's infra for detecting vector types for basic/pthread only 

yes. The difference is, hwloc code is independent of LLVM (and works on LLVM-less builds) - this code is not. If i moved this to hwloc, i'd have to either make it depend on LLVM there (i'd rather not), or copy some LLVM code that detects the cpu features. If you want this in hwloc code, i strongly prefer the latter (copy).

Reply:
Fair enough.